app/services/fetch_user_bid_ask_currency_service.rb

```diff
@@ -10,13 +10,39 @@ def call
     @user_currencies_service = user.user_currencies
   
     @user_currencies_service.each do |ucs|
-      buy_sell_currency = BidAskCurrency.find_by(currency_id: ucs['currency_id'])
+      if (buy_sell_currency = BidAskCurrency.find_by(currency_id: ucs['currency_id']) and ucs['purchase'].present?)
         
-      if(ucs['purchase'] == buy_sell_currency['bid'])
-        #BidAskMailer.send_bid_ask_email(@user)
-        p "tak"
-      else
-        p "nie"
+        @currency_name = buy_sell_currency.currency.name
+        @currency_bid = buy_sell_currency.bid
+        @currency_ask = buy_sell_currency.ask
+
+        #Checking purchase value
+        if(buy_sell_currency['bid'] <= ucs['purchase'])
+          @bid = SentEmail.find_by(user_id: user.id, currency_id: buy_sell_currency.currency_id ,operation_type: 'purchase')
+          unless @bid 
```

Comment:
Lepiej odwrócić ten warunek żeby nie było unless / else
```suggestion
          if @bid.present? && @bid.value > ucs['purchase']
```
i wtedy na dole @bid.update

Przy okazji `BidAskMailer.send_bid_email(@user, @currency_name, @currency_bid).deliver_now! `
Można wyjąć niżej żeby nie było 2 razy tego samego kodu